### PR TITLE
Metadata info protobuf format support

### PR DESCRIPTION
**Pull request by NRHelmi**
This PR adds support to the metadata info protobuf format.
```
cd RelationalAI.Examples
dotnet run ExecuteAsync --engine ENGINE --database DB --command "def output = 123"
```

### CODE REVIEW

File: RelationalAI.Protos/Proto/metadata.proto [hunk 1]
```diff
@@ -0,0 +1,160 @@
+syntax = "proto3";
```
Comment (denisgursky) [inline RelationalAI.Protos/Proto/metadata.proto:1]: See: https://github.com/RelationalAI/rai-sdk-javascript/pull/32#discussion_r932347814

**Review by larf311 — COMMENTED** [review]
Can you update the readme to include the fact that you need to install protobuf?

**Comment by larf311** [comment]
This doesn't seem to work as the accept header doesn't seem to be getting passed through. I added some log statements to the example:

```
POST https://login.relationalai.com/oauth/token
Content-Type: application/json
GET https://azure.relationalai.com:443/transactions/206c49ca-bdc7-c7c8-8edd-ad49a211455b/metadata
Key = Authorization, Value = Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlhaWVVsUTlhWm5sRGxZVFVfM05RTSJ9.eyJodHRwczovL3JlbGF0aW9uYWwuYWkvY2xhaW1zL2FjY291bnQiOiJyZWxhdGlvbmFsYWktdGVhbS1yZC11eCIsImlzcyI6Imh0dHBzOi8vbG9naW4ucmVsYXRpb25hbGFpLmNvbS8iLCJzdWIiOiIyOGVST3NsSU51WTBTZE15RFBOZ2xCbDJVRTAxR2txaEBjbGllbnRzIiwiYXVkIjoiaHR0cHM6Ly9henVyZS5yZWxhdGlvbmFsYWkuY29tIiwiaWF0IjoxNjU5NDY5NzEyLCJleHAiOjE2NTk0NzMzMTIsImF6cCI6IjI4ZVJPc2xJTnVZMFNkTXlEUE5nbEJsMlVFMDFHa3FoIiwic2NvcGUiOiJ1cGRhdGU6ZGF0YWJhc2UgbGlzdDpkYXRhYmFzZSBsaXN0OmNvbXB1dGUgcmVhZDpjb21wdXRlIGRlbGV0ZTpjb21wdXRlIGNyZWF0ZTpjb21wdXRlIHJ1bjp0cmFuc2FjdGlvbiIsImd0eSI6ImNsaWVudC1jcmVkZW50aWFscyJ9.RPXGxlTC1hnFRPym10vNBKWF9XskzKIKJH5UgUyxOrAzECD99tzZ_kWenBt-tbB5Zyr9a9S3Fb03eTyoDSMzxYyaHDPqZwuXc0ICxtVwAiyZPf8vQwJ8mwXCTnFqPMJinddgS3q57kpSyTlLEX3ch-GJu1ig_4sIz-XUol1lco46DfYgqdsM9pi8uB1kcTdPZHZMoF3lTPw7-VA96y5BJKE-L_3quPgDZatc23S8D1meBNG_kMliAWqLu-MElXDygmxDsg9RuKu8WgPLQr0s_BPsBk6JDsalpnmIUVDitYv56CWAeZ8mvPll0t_VCgP8-cTplkENP0iq0Jx_F6EceA
Content-Type: application/json
```

**Comment by NRHelmi** [comment]
> This doesn't seem to work as the accept header doesn't seem to be getting passed through. I added some log statements to the example:
> 
> ```
> POST https://login.relationalai.com/oauth/token
> Content-Type: application/json
> GET https://azure.relationalai.com:443/transactions/206c49ca-bdc7-c7c8-8edd-ad49a211455b/metadata
> Key = Authorization, Value = Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlhaWVVsUTlhWm5sRGxZVFVfM05RTSJ9.eyJodHRwczovL3JlbGF0aW9uYWwuYWkvY2xhaW1zL2FjY291bnQiOiJyZWxhdGlvbmFsYWktdGVhbS1yZC11eCIsImlzcyI6Imh0dHBzOi8vbG9naW4ucmVsYXRpb25hbGFpLmNvbS8iLCJzdWIiOiIyOGVST3NsSU51WTBTZE15RFBOZ2xCbDJVRTAxR2txaEBjbGllbnRzIiwiYXVkIjoiaHR0cHM6Ly9henVyZS5yZWxhdGlvbmFsYWkuY29tIiwiaWF0IjoxNjU5NDY5NzEyLCJleHAiOjE2NTk0NzMzMTIsImF6cCI6IjI4ZVJPc2xJTnVZMFNkTXlEUE5nbEJsMlVFMDFHa3FoIiwic2NvcGUiOiJ1cGRhdGU6ZGF0YWJhc2UgbGlzdDpkYXRhYmFzZSBsaXN0OmNvbXB1dGUgcmVhZDpjb21wdXRlIGRlbGV0ZTpjb21wdXRlIGNyZWF0ZTpjb21wdXRlIHJ1bjp0cmFuc2FjdGlvbiIsImd0eSI6ImNsaWVudC1jcmVkZW50aWFscyJ9.RPXGxlTC1hnFRPym10vNBKWF9XskzKIKJH5UgUyxOrAzECD99tzZ_kWenBt-tbB5Zyr9a9S3Fb03eTyoDSMzxYyaHDPqZwuXc0ICxtVwAiyZPf8vQwJ8mwXCTnFqPMJinddgS3q57kpSyTlLEX3ch-GJu1ig_4sIz-XUol1lco46DfYgqdsM9pi8uB1kcTdPZHZMoF3lTPw7-VA96y5BJKE-L_3quPgDZatc23S8D1meBNG_kMliAWqLu-MElXDygmxDsg9RuKu8WgPLQr0s_BPsBk6JDsalpnmIUVDitYv56CWAeZ8mvPll0t_VCgP8-cTplkENP0iq0Jx_F6EceA
> Content-Type: application/json
> ```

checking ... :eyes:

**Comment by larf311** [comment]
`dotnet run GetTransactionMetadataInfo --id 206c49ca-bdc7-c7c8-8edd-ad49a211455b`

**Comment by larf311** [comment]
It seems to be removing the `accept` header on line 218 of `Rest.cs`. Gotta run. Will check back later.

File: RelationalAI/TransactionAsyncResult.cs [hunk 2]
```diff
@@ -16,25 +16,25 @@
...
-    using Newtonsoft.Json;
+    using Relationalai.Protocol;
 
     public class TransactionAsyncResult : Entity
     {
         public TransactionAsyncCompactResponse Transaction { get; set; }
         public List<ArrowRelation> Results { get; set; }
-        public List<TransactionAsyncMetadataResponse> Metadata { get; set; }
+        public MetadataInfo Metadata { get; set; }
         public List<object> Problems { get; set; }
```
Comment (vilterp) [inline RelationalAI/TransactionAsyncResult.cs:26]: Let's follow this up with a PR similar to https://github.com/RelationalAI/rai-sdk-java/pull/16, introducing `GotCompleteResult`

**Comment by larf311** [comment]
Should we have the `schema.proto` in every repo? Seems like it should be centralized.

**Review by larf311 — COMMENTED** [review]
There are lots of warnings about closing braces on the Github diff. Should we address these? If not, where are they coming from?

**Comment by NRHelmi** [comment]
> There are lots of warnings about closing braces on the Github diff. Should we address these? If not, where are they coming from?

I don't really remember setting up something for csharp linting but indeed we need to do so
let me check if I can setup something :pray:

**Comment by NHDaly** [comment]
> Should we have the `schema.proto` in every repo? Seems like it should be centralized.

@larf311, agreed. There's an issue for it here: https://github.com/RelationalAI/rai-sdk-issues/issues/62

**Comment by NHDaly** [comment]
Yeah, what Helmi said. 👍

**Comment by NRHelmi** [comment]
if no other updates are required, I'll merge by the end of the day thank you

**Pull request merged**